Gems/Multiplayer/Code/Source/Components/SimplePlayerSpawnerComponent.cpp

```diff
@@ -72,35 +74,69 @@ namespace Multiplayer
+        {
+            AZ_Assert(false, "SimplePlayerSpawnerComponent has an out-of-bounds m_spawnIndex %i. Please ensure spawn index is always valid.", m_spawnIndex);
+            return AZ::Transform::Identity();
+        }
+
+        const AZ::EntityId spawnPointEntityId = m_spawnPoints[m_spawnIndex];
+        AZ::Transform spawnPointTransform;
+        AZ::TransformBus::EventResult(spawnPointTransform, spawnPointEntityId, &AZ::TransformInterface::GetWorldTM);
+        return spawnPointTransform;
+    }
+
+    const AZStd::vector<AZ::EntityId>& SimplePlayerSpawnerComponent::GetSpawnPoints() const
+    {
+        return m_spawnPoints;
+    }
+
+    uint32_t SimplePlayerSpawnerComponent::GetSpawnPointCount() const
+    {
+        return aznumeric_cast<uint32_t>(m_spawnPoints.size());
+    }
+
+    uint32_t SimplePlayerSpawnerComponent::GetNextSpawnPointIndex() const
+    {
         if (m_spawnPoints.empty())
         {
-            AZLOG_WARN("SimplePlayerSpawnerComponent is missing spawn points. Spawning new player at the origin.")
+            return static_cast<uint32_t>(-1);
         }
-        else
+
+        if (m_spawnIndex >= m_spawnPoints.size())
         {
-            m_spawnIndex %= m_spawnPoints.size();
-            const AZ::EntityId spawnPointEntityId = m_spawnPoints[m_spawnIndex];
-            
-            AZ::Entity* spawnPointEntity = nullptr;
-            AZ::ComponentApplicationBus::BroadcastResult(
-                spawnPointEntity, &AZ::ComponentApplicationBus::Events::FindEntity, spawnPointEntityId);
-            if (spawnPointEntity != nullptr)
-            {
-                transform = spawnPointEntity->GetTransform()->GetWorldTM();
-            }
-            else
-            {
-                AZLOG_WARN("The spawn point entity id for index %i is invalid. Spawning new player at the origin.", m_spawnIndex)
-            }
+            AZ_Assert(false, "SimplePlayerSpawnerComponent has an out-of-bounds m_spawnIndex %i. Please ensure spawn index is always valid.", m_spawnIndex);
+            return static_cast<uint32_t>(-1);
+        }
+
+        return m_spawnIndex;
+    }
+
+    void SimplePlayerSpawnerComponent::SetNextSpawnPointIndex(uint32_t index)
+    {
+        if (index >= m_spawnPoints.size())
+        {
+            AZLOG_WARN("SetNextSpawnPointIndex called with out-of-bounds spawn index %i; total spawn points: %i", index, aznumeric_cast<uint32_t>(m_spawnPoints.size()));
+            return;
         }
 
-        ++m_spawnIndex;
+        m_spawnIndex = index;
+    }
+
+    NetworkEntityHandle SimplePlayerSpawnerComponent::OnPlayerJoin(
+        [[maybe_unused]] uint64_t userId, [[maybe_unused]] const MultiplayerAgentDatum& agentDatum)
+    {
+        const PrefabEntityId prefabEntityId(AZ::Name(m_playerSpawnable.m_spawnableAsset.GetHint().c_str()));
+
+        const AZ::Transform transform = GetNextSpawnPoint();
+        m_spawnIndex = ++m_spawnIndex % m_spawnPoints.size();
```

Comment:
nit: Do we think this is thread safe? I assume the worst that can happen is multiple players spawning on same point?

Reply:
Yeah, I guess players would spawn in on the same point. We're spawning on the main thread, but yeah, maybe it's a problem if we move handling network traffic to a new thread.